pages/Identity/Overview.md

```diff
@@ -7,22 +7,319 @@ menu: Identity
+     * @param endBlock Block number considered to be the end of the delegate permissions
+     */
+    event DSNPRemoveDelegate(address delegate, uint64 endBlock);
+
+    /**
+     * @dev Add Delegate (default permissions determined by implementation)
+     * @param newDelegate Address to delegate new permissions to 
+     * 
+     * MUST be called by the owner
+     * MUST consider newDelegate to be valid from the beginning to time
+     * MUST emit DSNPAddDelegate
+     */
+    function delegate(address newDelegate) external;
+
+    /**
+     * @dev Add Delegate By EIP-712 Signature (default permissions determined by implementation)
+     * @param r ECDSA Signature r value
+     * @param s ECDSA Signature s value
+     * @param v EIP-155 calculated Signature v value
+     * @param newDelegate Address to delegate new permissions to
+     * 
+     * MUST be signed by the owner
+     * MUST consider newDelegate to be valid from the beginning to time
+     * MUST emit DSNPAddDelegate
+     */
+    function delegateBySignature(bytes32 r, bytes32 s, uint32 v, address newDelegate) external;
+
+    /**
+     * @dev Remove delegate
+     * @param delegate Address to remove all permissions from
+     * @param endBlock Block number to consider the permissions terminated (MUST be > 0x0). 
+     * 
+     * MUST be called from owner or the delegate  
+     * MUST store endBlock for response in isAuthorizedToAnnounce
+     * MUST emit DSNPRemoveDelegate
+     */
+    function delegateRemove(address delegate, uint64 endBlock) external;
+
+    /**
+     * @dev Remove delegate By EIP-712 signature (default permissions determined by implementation)
+     * @param delegate Address to remove all permissions from
+     * @param endBlock Block number to consider the permissions terminated (MUST be > 0x0).
+     * @param r ECDSA Signature r value
+     * @param s ECDSA Signature s value
+     * @param v EIP-155 calculated Signature v value
+     * 
+     * MUST be signed by owner or by the delegate
+     * MUST store endBlock for response in isAuthorizedToAnnounce
+     * MUST emit DSNPRemoveDelegate
+     */
+    function delegateRemoveBySignature(bytes32 r, bytes32 s, uint32 v, address delegate, uint64 endBlock) external;
+
+    /**
+     * @dev Checks to see if address is authorized to announce messages
+     * @param addr Address that is used to test against erecover
+     * @param blockNumber Check for authorization at a particular block number
+     * @return boolean 
+     * 
+     * @dev Return MAY change as deauthorization can revoke past messages
+     */
+    function isAuthorizedToAnnounce(address addr, uint64 blockNumber) external view returns (bool);
+}
+```
+
+### Option 2: With Permissions
+
+Specific permissions are granted when the delegate is added.
+
+```solidity
+/**
+ * @dev DSNP Identity Interface for managing delegates
+ */
+interface IDelegation {
+
+    /**
+     * @dev Should we do a Bitwise permission system instead of enum?
```

Comment:
Discussed in a meeting:

> Should we do a Bitwise permission system instead of enum?

While the majority of developers desired a bitwise permission system, the cost of data storage and the limited ability to predict the needed permissions into the future lead toward an enum system.

Bitwise would allow for many combinations that just do not make sense, taking up space.

Enums still allows for a bitwise implementation, but present a better interface for the users, specifically for the validators calling `isAuthorizedTo`.